pkg/service/redisstore.go

```diff
@@ -97,74 +97,66 @@ func (s *RedisStore) Stop() {
 	}
 }
 
-func (s *RedisStore) StoreRoom(_ context.Context, room *livekit.Room) error {
+func (s *RedisStore) StoreRoom(_ context.Context, room *livekit.Room, internal *livekit.RoomInternal) error {
 	if room.CreationTime == 0 {
 		room.CreationTime = time.Now().Unix()
 	}
 
-	data, err := proto.Marshal(room)
+	roomData, err := proto.Marshal(room)
 	if err != nil {
 		return err
 	}
 
-	pp := s.rc.Pipeline()
-	pp.HSet(s.ctx, RoomsKey, room.Name, data)
-
-	if _, err = pp.Exec(s.ctx); err != nil {
-		return errors.Wrap(err, "could not create room")
-	}
-	return nil
-}
-
-func (s *RedisStore) StoreRoomInternal(_ context.Context, roomName livekit.RoomName, internal *livekit.RoomInternal) error {
-	data, err := proto.Marshal(internal)
-	if err != nil {
-		return err
+	var internalData []byte
+	if internal != nil {
+		internalData, err = proto.Marshal(internal)
+		if err != nil {
+			return err
+		}
 	}
 
 	pp := s.rc.Pipeline()
-	pp.HSet(s.ctx, RoomInternalKey, roomName, data)
+	pp.HSet(s.ctx, RoomsKey, room.Name, roomData)
+	if internalData != nil {
+		pp.HSet(s.ctx, RoomInternalKey, room.Name, internalData)
+	}
 
 	if _, err = pp.Exec(s.ctx); err != nil {
 		return errors.Wrap(err, "could not create room")
 	}
 	return nil
 }
 
-func (s *RedisStore) LoadRoom(_ context.Context, roomName livekit.RoomName) (*livekit.Room, error) {
-	data, err := s.rc.HGet(s.ctx, RoomsKey, string(roomName)).Result()
+func (s *RedisStore) LoadRoom(_ context.Context, roomName livekit.RoomName, includeInternal bool) (*livekit.Room, *livekit.RoomInternal, error) {
+	roomData, err := s.rc.HGet(s.ctx, RoomsKey, string(roomName)).Result()
```

Comment:
There is a chance you may get inconsistent Room and RoomInternal data if a write happens between the 2 HGET. Is there any reason not to use a pipeline here?